### PR TITLE
feat(quantic):  QuanticSmartSnippetSource created

### PR DESCRIPTION
**Pull request by mmitiche**
[SFINT-4796](https://coveord.atlassian.net/browse/SFINT-4796
)
- The Quantic Smart Snippet Source component created.
- The Quantic Smart Snippet Source displays the clickUri and the title of the smart snippet source.
- The Quantic Smart Snippet Source binds the correct actions to the events triggered  on the links within  this component in order to log analytics.

#### Design doc:
[Design-doc__Quantic-Smart-Snippet-components](https://docs.google.com/document/d/1kJswWu6qHp5ExMHQagkrCWphrjRmys8a5tQ0oAnQkJU/edit)

#### Demo:

https://user-images.githubusercontent.com/86681870/216156849-86e1fca3-2f2c-408b-bb5e-2c9d4d0dd5f8.mov

[SFINT-4796]: https://coveord.atlassian.net/browse/SFINT-4796?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js [hunk 1]
```diff
@@ -131,6 +131,34 @@ export class ResultUtils {
...
 }
 
+export class LinkUtils {
+  /**
+   * Binds the logging of a link
+   * @returns An unbind function for the events
+   * @param {HTMLAnchorElement} link the link element
+   * @param {{select:function, beginDelayedSelect: function, cancelPendingSelect: function  }} actions
+   */
+  static bindAnalyticsToLink(link, actions) {
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js:141]: My brain likes this util very much

File: packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    const snippetSourceUrlElement = this.template.querySelector(
+      '.smart-snippet__source-uri'
+    );
+    /** @type {HTMLAnchorElement} */
+    const snippetSourceTitleElement = this.template.querySelector(
+      '.smart-snippet__source-title'
+    );
+
+    LinkUtils.bindAnalyticsToLink(snippetSourceUrlElement, this.actions);
+    LinkUtils.bindAnalyticsToLink(snippetSourceTitleElement, this.actions);
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.js:52]: I noticed the util returns a function that will unbind the events. Is that not something we want keep a reference to and execute on `disconnectedCallback`?

File: packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.css [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.smart-snippet__source-uri {
+  color: #1a1b1e;
+  font-weight: 400;
+}
+
+.smart-snippet__source-title {
+  line-height: 28px;
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.css:7]: would you mind filling me in on why we need this specific line-height?

File: packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.js (hunk 2 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.js:52]: Very true, I'll update this 👌

File: packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.css (hunk 3 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.css:7]: This specific line-height was used in the mockups and it's different than default line height that comes with the SLDS classes, do you think we should only stick to the default styling of the SLDS text classes?

File: packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<template>
+  <div class="smart-snippet__source">
+    <div>
+      <a class="slds-text-title_bold smart-snippet__source-uri" href={uri}>{uri}</a>
+    </div>
+    <div>
+      <a class="smart-snippet__source-title slds-text-heading_small" href={uri}>{title}</a>
```
[the remark this answers is not in the document]
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.html:7]: Probably was discussed when this component was first invented years ago in the JSUI x)  here are the mockups of the smart snippet if you wanna take a closer look: https://www.figma.com/file/KPJhWCQ7LubfM3RzjTWDb9/Smart-Snippets---Quantic?node-id=31%3A1355&t=zPGVil6VFsvjhMpR-0
Personally, it feels natural to me to have the uri in a slightly smaller font size on top of the title.

File: packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js (hunk 1 above)
Comment (SimonMilord) [reply packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js:141]: very clean!

File: packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.css (hunk 3 above)
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.css:7]: Maybe it's just me but I prefer using margin and padding over altering line-height. In my experience line-height and font-size have a tight relationship and given font-size is something we have less control over unless specified, I'm cautious about it. Maybe overly cautious.

**Review by SimonMilord — APPROVED** [review]
Looks good, the util is 
image

**Review by lbergeron — APPROVED** [review]
Nice!

File: packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.css (hunk 3 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.css:7]: Makes sense👌 I'll update this and use the SLDS padding classes instead.

**Pull request merged**